### ISSUE TITLE
Bugs and improvement

### ISSUE THREAD

**Issue by MSouilah** [AdrianVillamayor/Wizard-JS#1]
Hi,
First of all, thank you for this very good wizard I were looking for to replace the jQuery Steps plugin I use.
I have founded some bugs and I have some improvements to propose.

Bugs:
Fields prev, next and finish not working due to:
            wz_next: (args != undefined && args.hasOwnProperty("next")) ? args.wz_next : ".next",
            wz_prev: (args != undefined && args.hasOwnProperty("prev")) ? args.wz_prev : ".prev",
            wz_finish: (args != undefined && args.hasOwnProperty("prev")) ? args.wz_prev : ".finish",
to be replaced by
            wz_next: (args != undefined && args.hasOwnProperty("wz_next")) ? args.wz_next : ".next",
            wz_prev: (args != undefined && args.hasOwnProperty("wz_prev")) ? args.wz_prev : ".prev",
            wz_finish: (args != undefined && args.hasOwnProperty("wz_finish")) ? args.wz_prev : ".finish",

is_form is not working as expected. When turn to false, the next button still waiting for required fields, I think it will be better to overpassed the required fields when using this. I have tested on my fork and it works well.

`<div class="wizard-step" data-type="form">` when using default as data-type not working either.

wizard.vertical did not place wizard-nav and wizard-step side by side but first wizard-nav and then wizard- step. It will be great if the two could be side by side.

Improvements:
I create a blink mode (true/false) to show highlight-error like you do or to stay red until the field is filled. I have also passed highlight-valid to show green fields correctly fields. I can make a pull requests if you want (I need to learn how to pull requests first !!)

Color steps passing by argument for the progress mode (wz_nav_style)

Question:
I can not find if it possible or not to prevent next step not only by required fields (checkForm) but also with some condition to fulfil ? Maybe using the `document.addEventListener("nextWizard", function (e) {});` but I do not know which parameters to turn to false or break or something else ?

**Comment by AdrianVillamayor** [comment]
Hii @MSouilah,

I have uploaded a fix for the fields issue. Besides I have changed the vertical system by one of columns (side by side), as you indicate in your comment.

On the other hand, I am working on a step blocking function and checking the is_form bug, as soon as I have it I will push it and update the version.

Finally, the project is open to any improvement, feel free to upload your changes. Always in a new branch and then we make a merge to the main one, to avoid conflicts.

Thank you very much for your comment and the help !

**Comment by AdrianVillamayor** [comment]
Hii @MSouilah ,

Lack of tab blocking system, I have updated the entire model related to form management. It is now more intuitive and easier to manage. 

You have all the changes [here](https://github.com/AdrianVillamayor/Wizard-JS/compare/v1.6...v1.7)

**Comment by AdrianVillamayor** [comment]
Hii @MSouilah ,

All bugs have been fixed and I have added: 
 - A method to lock the wizard in one step.
 - A method to unlock the wizard in one step.
 - A method to restart the wizard in a comfortable and easy way.

You have all the changes [here](https://github.com/AdrianVillamayor/Wizard-JS/compare/v1.7.5...v1.7.6) and explained in the [README](https://github.com/AdrianVillamayor/Wizard-JS/blob/main/README.md).

Any other bug or idea don't hesitate to comment it 🙌🏻

Thank you very much for your comments and I hope I have fixed them all !
